pkg/scanners/scannerv4/convert.go

```diff
@@ -186,55 +188,116 @@ func vulnerabilities(vulnerabilities map[string]*v4.VulnerabilityReport_Vulnerab
 	return vulns
 }
 
-func setScoresAndScoreVersion(vuln *storage.EmbeddedVulnerability, vulnCVSS *v4.VulnerabilityReport_Vulnerability_CVSS) error {
-	if vulnCVSS == nil {
+func setScoresAndScoreVersions(vuln *storage.EmbeddedVulnerability, cvssMetrics []*v4.VulnerabilityReport_Vulnerability_CVSS) error {
+	if len(cvssMetrics) == 0 {
 		return nil
 	}
-	errList := errorhelpers.NewErrorList("failed to parse vector")
-	if v2 := vulnCVSS.GetV2(); v2 != nil {
-		if c, err := cvssv2.ParseCVSSV2(v2.GetVector()); err == nil {
-			err = cvssv2.CalculateScores(c)
-			if err != nil {
-				errList.AddError(fmt.Errorf("calculating CVSS v2 scores: %w", err))
+
+	errList := errorhelpers.NewErrorList("failed to get CVSS Metrics")
+	var scores []*storage.CVSSScore
+	for _, cvss := range cvssMetrics {
+		score := &storage.CVSSScore{
+			Source: CVSSSource(cvss.Source),
+			Url:    cvss.GetUrl(),
+		}
+		hasV3 := false
+		if cvss.GetV3() != nil {
+			baseScore, cvssV3, v3Err := toCVSSV3Scores(cvss, vuln.GetCve())
+			if v3Err == nil && cvssV3 != nil {
+				score.CvssScore = &storage.CVSSScore_Cvssv3{Cvssv3: cvssV3}
+				scores = append(scores, score)
+				// CVSS metrics has maximum two entries, one from NVD, one from Rox updater if available
+				if len(cvssMetrics) == 1 || (len(cvssMetrics) > 1 && cvss.Source != v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_NVD) {
+					vuln.ScoreVersion = storage.EmbeddedVulnerability_V3
+					vuln.Cvss = baseScore
+					vuln.CvssV3 = cvssV3
+					hasV3 = true
+				}
+			} else {
+				errList.AddError(v3Err)
 			}
-			// Use the report's score if it exists.
-			if baseScore := v2.GetBaseScore(); baseScore != 0.0 && baseScore != c.Score {
-				log.Debugf("Calculated CVSSv2 score does not match given base score (%f != %f) for %s. Using given score...", c.Score, baseScore, vuln.GetCve())
-				c.Score = baseScore
+		}
+
+		if cvss.GetV2() != nil {
+			baseScore, cvssV2, v2Err := toCVSSV2Scores(cvss, vuln.GetCve())
+			if v2Err == nil && cvssV2 != nil {
+				score.CvssScore = &storage.CVSSScore_Cvssv2{Cvssv2: cvssV2}
```

Comment:
Can a vulnReport have both V2 and V3 CVSS scores? If so, this block which is after the `if cvss.GetV3() != nil` block would mean that V2 score will overwrite V3 score . Is that how we want to prioritize when both scores are present?

Reply:
Make sense. Not sure why the tests didn't catch this. I'll take a look.